Implemented Stack Calculator.

LightnessOfBeing opened the pull request.

LightnessOfBeing committed:
Implemented Stack Calculator.

LightnessOfBeing committed:
Added non-significant changes.

yurii-litvinov commented on StackCalculator/src/main/java/ru/academic/university/mit/vishnyakov/Stack.java:
```diff
@@ -0,0 +1,116 @@
+     */
+
+    private class Node {
+        private T T;
```
Мм, странное имя поля

yurii-litvinov commented on StackCalculator/src/main/java/ru/academic/university/mit/vishnyakov/Stack.java:
```diff
@@ -0,0 +1,116 @@
+     * Class that represents nodes.
+     */
+
+    private class Node {
```
static?

yurii-litvinov commented on StackCalculator/src/test/java/ru/academic/university/mit/vishnyakov/StackTest.java:
```diff
@@ -0,0 +1,68 @@
+public class StackTest {
+    @Test
+    public void isEmptyNoElements() throws Exception {
+        Stack<Double> stack = new Stack<>();
```
Можно сделать его полем и инициализировать в методе с аннотацией `@Before`, чтобы в каждом тесте это не писать

LightnessOfBeing committed:
Added Before annotaion in tests, renamed field in node class, added static modifier.

yurii-litvinov commented on StackCalculator/src/test/java/ru/academic/university/mit/vishnyakov/StackTest.java:
```diff
@@ -1,68 +1,72 @@
 public class StackTest {
+
+    Stack<Double> stackDouble;
+    Stack<Integer> stackInteger;
```
Но зачем они пакетные?

yurii-litvinov commented on StackCalculator/src/main/java/ru/academic/university/mit/vishnyakov/Stack.java:
```diff
@@ -85,7 +85,7 @@ public T pop() throws NoSuchElementException {
         }
         assert first != null;
-        T e = first.T;
+        T e = (T) first.element;
```
Но first же тоже можно было сделать не сырым Node, а генериковым

yurii-litvinov reviewed (approved):
Не то, чтобы то, что я хотел, но ладно, зачтена
